pkg/config/config.go

```diff
@@ -129,3 +165,38 @@ func (gc *Config) configureProxies(httpProxy, httpsProxy, noProxy string) {
 		os.Setenv(strings.ToUpper(noProxyEnvVar), noProxy)
 	}
 }
+
+func addSecretDataToConfig(config *Config, secret *v1.Secret) {
+	extractedValues := map[string]string{}
+	keys := []string{caCertsKey, httpProxyKey, httpsProxyKey, noProxyKey, skipTLSVerifyKey}
```

Comment:
might as well convert all keys. no need to filter out by specific key.

Reply:
@joe-kimmel-vmw Any opinions on this? I can see pros and cons for either approach.

Mainly, my thought is that we should not have it process unknown fields at all.

However, I see the benefit in maintaining this long term as far as not needing to maintain this list of keys in this func. In `addDataToConfig`, we are already ignoring unknown fields.

Reply:
After discussing further, we will go with approach of just iterating over all key-value pairs. 

This has been addressed in 68f0cf1.